### DISCUSSION TITLE
Codespaces + Live share can lose all work

### DISCUSSION THREAD

**Discussion by samskiter** [community/community#55771] [Codespaces]
### Select Topic Area

Bug

### Body

New to codespaces and loving it. Tried using Live Share with it - which worked like a dream to collab with a colleague.

After he left the live share session, I kept working, finishing some stuff off. I got some stuff working and left things to show him later. I think I may have switched back to another codespace around this time.

I now can't seem to find all the work I did. When I open one of my codespaces I get an error that the Workspace does not exist!

Looking through the File -> Open Recent menu, it looks like the way live share works is it creates a new workspace in /tmp (little did I realise!)

I think this means that when my Codespace is shut down I lose all of my work? I assumed we were collaborating on my open workspace, not a copy of it!

Attaching a couple of images.

![Screenshot 2023-05-17 at 20 41 20](https://github.com/community/community/assets/1271643/b299988d-4aaf-4151-a3c4-2531aa9a07d7)
![Screenshot 2023-05-17 at 20 41 09](https://github.com/community/community/assets/1271643/cd6f746f-a789-4037-8a58-ead35292b840)

**Comment by samruddhikhandale** [comment]
Hi 👋 

Thanks for reporting the issue and apologies for the inconvenience. 

> Looking through the File -> Open Recent menu, it looks like the way live share works is it creates a new workspace in /tmp (little did I realise!)

This is not an expected behavior, the changes should continue to exist in the codespaces' workspace. 

> I think this means that when my Codespace is shut down I lose all of my work? I assumed we were collaborating on my open workspace, not a copy of it!

Changes to the `/workspaces/*` folder remain intact for the entire life of a codespace, and they won't be lost after stopping and restarting a codespace. 

I suspect an issue with Liveshare which caused the changes to get saved to the `/tmp/vsliveshare-workspace-tmp-*` folder. `/tmp` is a temporary folder and changes are not saved across resumes.

@samskiter do you mind opening a ticket for Live Share, it looks like a bug on their end which directly switched to use `/tmp` folder instead of the current (`/workspaces/*) folder. 

Also, I tried to reproduce Codespaces + Live share: For me, everything was saved on the `/workspaces` folder which persisted even after stopping and restarting the codespace.

**Reply by samskiter** [reply]
thanks for the reply @samruddhikhandale I've done a bit more playing and I think I may have been mistaken but I think there is still a bug...

So I think what happened was I did LiveSharing in both directions (I shared with someone else and they shared with me). When they shared with me, it opened VSCode and switched the workspace to the /tmp/blah directory. This switch got persisted so when I re-opened my codespace I was taken to the liveshare directory, not my workspace for that codespace. It seems that when you open someone's liveshare while you also have a codespace open, it reuses something from that codespace and changes your workspace directory for that codespace.

The fix after opening the codespace and finding yourself in this situation is to open the correct workspace. Then after your restart your codespace again, you will be in the correct workspace.

So I think the minimal instructions to reproduce are:

1. Open your codespace in VSCode desktop
2. Join a liveshare session
3. Leave close VSCode desktop and have the liveshare session shut down
4. Re-launch your own codespace - you should receive an error that the workspace can't be found

There's a general bit of jank with liveshare still but I think this repros it
